app/settings/server/functions/settings.ts

```diff
@@ -0,0 +1,364 @@
+import { Meteor } from 'meteor/meteor';
+import _ from 'underscore';
+
+import { SettingsBase, SettingValue } from '../../lib/settings';
+import SettingsModel from '../../../models/server/models/Settings';
+
+const blockedSettings = new Set<string>();
+const hiddenSettings = new Set<string>();
+
+if (process.env.SETTINGS_BLOCKED) {
+	process.env.SETTINGS_BLOCKED.split(',').forEach((settingId) => blockedSettings.add(settingId.trim()));
+}
+
+if (process.env.SETTINGS_HIDDEN) {
+	process.env.SETTINGS_HIDDEN.split(',').forEach((settingId) => hiddenSettings.add(settingId.trim()));
+}
+
+const overrideSetting = (_id: string, value: SettingValue, options: ISettingAddOptions): SettingValue => {
+	const envValue = process?.env?.[_id];
+	if (envValue) {
+		if (envValue.toLowerCase() === 'true') {
+			value = true;
+		} else if (envValue.toLowerCase() === 'false') {
+			value = false;
+		} else if (options.type === 'int') {
+			value = parseInt(envValue);
+		}
+		options.processEnvValue = value;
+		options.valueSource = 'processEnvValue';
+	} else if (typeof Meteor.settings?.[_id] !== 'undefined') {
```

Comment:
`Meteor.settings` is always a defined object (https://docs.meteor.com/api/core.html#Meteor-settings)
```suggestion
	} else if (typeof Meteor.settings[_id] !== 'undefined') {
```